File: app/schemas/ocorrencia.py
"""Pydantic schemas for Ocorrencia entity."""

from datetime import datetime

from pydantic import BaseModel

from app.models.enums import Severidade


class OcorrenciaBase(BaseModel):
    missao_id: int
    tipo_ocorrencia_id: int
    descricao: str
    severidade: Severidade


class OcorrenciaCreate(OcorrenciaBase):
    pass


class OcorrenciaResponse(OcorrenciaBase):
    id: int
    registrada_por: int
    registrada_em: datetime
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/ocorrencia.ts
/** TypeScript interfaces for Ocorrencia entity matching Pydantic schemas. */

import type { Severidade } from './enums';

export interface OcorrenciaResponse {
  id: number;
  missao_id: number;
  tipo_ocorrencia_id: number;
  descricao: string;
  severidade: Severidade;
  registrada_por: number;
  registrada_em: string;
  created_at: string;
  updated_at: string;
}

export interface OcorrenciaCreate {
  missao_id: number;
  tipo_ocorrencia_id: number;
  descricao: string;
  severidade: Severidade;
}
